core-api/src/services/permisson.service.ts

```diff
@@ -0,0 +1,84 @@
+import db from "../config/database";
+import { Permisson, SafePermisson } from "../entities/permission.entity";
+import { ServerError, ErrorCode } from "../utils/serverError";
+import { ValidationError, validate } from "class-validator";
+import Messages from "../utils/messages";
+
+/**
+ * @namespace Services
+ * @class PermissonService
+ */
+export class PermissonService {
+    /**
+     * Create permisson
+     * 
+     * @param name
+     * @param email
+     * @param password
+     */
+    public static async create(name: string): Promise<SafePermisson | Error> {
+        const repository = db.getRepository(Permisson);
+        const newPermisson = new Permisson(name);
+
+        await this.validateFields(newPermisson);
+
+        try {
+            await repository.save(newPermisson)
+            return newPermisson.toSafe();
+        } catch (e) {
+            throw new ServerError(e.message, ErrorCode.DATABASE_CONSTRAINT);
+        }
+    }
+
+    private static async validateFields(newPermisson: Permisson): Promise<void> {
+        const validationErrors: ValidationError[] = await validate(newPermisson);
+
+        if (validationErrors.length > 0) {
+            const errorMessages = validationErrors.map((err) => Object.values(err.constraints));
+
+            throw new ServerError(errorMessages.join(". "), ErrorCode.FIELD_VALIDATION);
+        }
+
+        if (await this.permissonExists(newPermisson.getName())) {
+            throw new ServerError(Messages.validation.username_exists, ErrorCode.ALREADY_EXISTS);
```

Comment:
Create a new message for this case instead of using `validation.username_exists`.

Reply:
Done